src/components/Favorites.tsx

```diff
@@ -0,0 +1,87 @@
+import { useSearchParams } from 'react-router-dom';
+import styled from 'styled-components';
+
+import { useAppContext } from '../context/appContext';
+import { useCarsQuery } from '../generated/graphql';
+import { Cars } from '../generated/graphql';
+import { getVariablesQueryCars } from '../utils/getVariablesQueryCars';
+import CarCard from './CarCard';
+import Description from './Description';
+import Filters from './Filters';
+import Spinner from './Spinner';
+
+export interface CarItem extends Cars {
+  isFavorite: boolean;
+}
+
+const FavoritesContainer = styled.div`
+  background-color: ${(props) => props.theme.colors.neutralColor};
+  border-radius: 0.5rem;
+  flex: 1;
+  display: flex;
+  flex-direction: column;
+`;
+
+const Error = styled.p`
+  text-align: center;
+  font-weight: bold;
+  font-size: 2rem;
+`;
+
+const NotFound = styled.p`
+  color: ${(props) => props.theme.colors.successColor2};
+  font-weight: bold;
+  font-size: 2rem;
+  text-align: center;
+`;
+
+const Favorites = () => {
+  const { user, favorites } = useAppContext();
+  const [search] = useSearchParams();
+  const { orderBy, whereCars } = getVariablesQueryCars(search, user?.id);
+
+  const {
+    loading: loadingCars,
+    data: dataCars,
+    error: errorCars,
+  } = useCarsQuery({
+    variables: {
+      orderBy,
+      whereCars: {
+        ...whereCars,
+        id: {
+          _in: favorites,
+        },
+      },
+    },
+  });
+
+  return (
+    <>
+      <Filters />
+      <FavoritesContainer>
+        <Description />
+        {errorCars && <Error>There was an error</Error>}
```

Comment:
it seems like you have a `Error` component on each page, please try to create a single component and reuse it

Reply:
I solved it